apps/client/src/features/viewers/timer/Timer.tsx

```diff
@@ -61,29 +63,62 @@ export default function Timer(props) {
   const clock = formatTime(time.clock, formatOptions);
   const showOverlay = pres.text !== '' && pres.visible;
   const isPlaying = time.playback !== 'pause';
-  const normalisedTime = time.current === null ? null : Math.max(time.current, 0);
+  const isNegative =
+    (time.current ?? 0) < 0 && time.timerType !== TimerType.Clock && time.timerType !== TimerType.CountUp;
+
+  const showEndMessage = time.current < 0 && general.endMessage;
   const showProgress = time.playback !== 'stop';
+  const showFinished = time.finished && (time.timerType !== TimerType.Clock || showEndMessage);
+  const showClock = time.timerType !== TimerType.Clock;
   const baseClasses = `stage-timer ${isMirrored ? 'mirror' : ''}`;
 
+  const stageTimer = getTimerByType(time);
+
+  let display = '';
+
+  if (typeof stageTimer === 'string') {
+    display = stageTimer;
+  } else if (stageTimer === null || typeof stageTimer === 'undefined' || isNaN(stageTimer)) {
+    display = '-- : -- : --';
+  } else {
+    display = formatDisplay(millisToSeconds(stageTimer), true);
+  }
+
+  if (isNegative) {
+    display = `-${display}`;
+  }
+
+  const stageTimerCharacters = display.replace('/:/g', '').length;
+
   return (
-    <div className={time.finished ? `${baseClasses} stage-timer--finished` : baseClasses} data-testid='timer-view'>
+    <div className={showFinished ? `${baseClasses} stage-timer--finished` : baseClasses} data-testid='timer-view'>
       <NavigationMenu />
       <div className={showOverlay ? 'message-overlay message-overlay--active' : 'message-overlay'}>
         <div className='message'>{pres.text}</div>
       </div>
 
-      <div className='clock-container'>
+      <div
+        className='clock-container'
+        style={{
+          opacity: `${showClock ? 1 : 0}`,
+        }}
```

Comment:
I would prefer that we do this with a classname

something like
className={`clock-container ${showClock ? '' : 'hidden'}`}

Your solution is fine, and the performance impact is negligible. But currently we define all the styles in the CSS, and it is best to maintain some consistency to help our future selves with maintenance